internal/group/forum/usecase.go

```diff
@@ -0,0 +1,106 @@
+package forum
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/np-inprove/server/internal/entity"
+	"github.com/np-inprove/server/internal/entity/forum"
+	"github.com/np-inprove/server/internal/entity/group"
+)
+
+type UseCase interface {
+	ListPrincipalForums(ctx context.Context, principal string) ([]*entity.Forum, error)
```

Comment:
Considering prerequisite for forum is existence of a group, I think it is apt to have first arg be short name of group

```suggestion
	ListForums(ctx context.Context, principal, shortName string) ([]*entity.Forum, error)
```

Reply:
Same for below. For Create and Update can have shortName and forumShortName. I think should indicate in comments clearly what the args refer to

Reply:
I'm not sure how to change the httphandler file to accommodate the changes to the args. Could use some help